### DISCUSSION TITLE
Gen 5 SSL cert questions

### DISCUSSION THREAD

**Discussion by ZeikkuSSJ7** [mm201/pkmnFoundations#115] [General]
Hi, for the past few days I have been setting up an dwc server with this framework following the old guide at the wiki, and up until now I have been able to do everything in Gen 4 (gts, battle videos, mystery gifts), but with gen 5 I can only get mystery gifts to work. 
I took a look at the code and I discovered that Gen 5 uses SSL for some kind of handshake between the server and the game, but I have no idea as to which of the domains are used for the SSL cert, nor how do I set it up. The errors that I am getting are:

- 13275 (sending a Pokemon to the GTS)
- 10607 (trying to search videos on the battle server)

My games (Platinum, HeartGold, Black, Black 2) come from my original copies, and I patched them with WfcPatcher. My testing devices were:

- An R43DS flashcard with YSMenu
- melonDS, because of its WI-Fi capabilities
- nds-bootstrap, but it crashed with Gen 5, probably due to antipiracy patches not working well with WfcPatcher (Edit: with the latest versión now it doesn't crash, but same situation).

There are two things to consider with my setup. First, I am using a Windows machine as the server with IIS pages enabled and all the requests to the GTS are going through the Apache reverse proxy, and second, I had to update this project to .NET 4.8 to get it to even open in Visual Studio 2022, but aside from that everything else works.

As for the Battle Video Server, everything works for Gen 4, but in Gen 5 the console throws an exception related to the SSL handshake I was talking about with either of this messages.

> A call to SSPI failed, see inner exception.
> The client and server cannot communicate, because they do not possess a common algorithm.

So, is there anyone that knows exactly what the issue is? Do I have to enable SSLv3 in my server to make it work? And how do I exactly create the certificate for the server, just by using any CA like LetsEncrypt or like the nds-constraint project?
Thanks for any help!

**Comment by mm201** [comment]
13275 happens when it can't communicate with `pkvldtprod.nintendo.co.jp` which is the anti-cheat server. If you're using noSSL patches, you should just be able to add a DNS entry for it, pointing to the GTS, and it will work.

The SSPI failed exception is because the DS is trying to communicate with the Battle Video server over SSLv3 with an RC4 cipher. Check the comment at: https://github.com/mm201/pkmn-classic-framework/blob/a9ca79f6bcdff3d28977de96976b583306069521/GlobalTerminalService/GTServerBase.cs#L215-L220
You can manually enable the ciphers through the registry or with [IISCrypto](https://www.nartac.com/Products/IISCrypto/Download).

**Reply by mm201** [reply]
Oh and literally any certificate will work. The game doesn't care. Just keep using the dummy for testing if it's easier.

**Reply by ZeikkuSSJ7** [reply]
Okay, finally everything works. My god this was REALLY hard to set up, just because very little things were messing around inside the project. Thank you for your help!

PS: if anyone gets stuck while trying to set this up for themselves, don't hesitate to ask me. It was a journey, but with the correct guidance it can be a breeze.

**Reply by ZeikkuSSJ7** [reply]
And yeah, the pkvldtprod domain I figured it out by setting a DNS to your pkvldtprod IP. I just thought I didn't need anticheat just because I wasn't battling anyone nor using Game Sync, which to be honest, was really dumb.
